src/porepy/models/boundary_condition.py

```diff
@@ -0,0 +1,80 @@
+import porepy as pp
+
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+from typing import Callable, Sequence
+
+
+class BoundaryConditionMixin(ABC):
+    """TODO"""
+
+    mdg: pp.MixedDimensionalGrid
+    """Mixed-dimensional grid for the current model. Normally defined in a mixin
+    instance of :class:`~porepy.models.geometry.ModelGeometry`.
+
+    """
+
+    domain_boundary_sides: Callable[[pp.Grid], pp.domain.DomainSides]
+    """Boundary sides of the domain. Normally defined in a mixin instance of
+    :class:`~porepy.models.geometry.ModelGeometry`.
+
+    """
+
+    time_manager: pp.TimeManager
+    """Time manager. Normally set by an instance of a subclass of
+    :class:`porepy.models.solution_strategy.SolutionStrategy`.
+
+    """
+
+    units: "pp.Units"
+    """Units object, containing the scaling of base magnitudes."""
+
+    @abstractmethod
+    def update_boundary_conditions(self) -> None:
```

Comment:
@Yuriyzabegaev  The naming of this is very dangerous, since only on `s` distinguishes it from the part where there is an actual implementation to be used in child classes.

Reply:
@vlipovac I agree. What about `update_all_boundary_conditions`?

Reply:
I included that refactoring in the changes I am about to make with the energy balance